### PR TITLE
Clean up std primitive docs

### PR DESCRIPTION
**Pull request by brson**
This makes the primitive descriptions on the front page read properly
as descriptions of types and not of the associated modules.

Having the primitive and module docs derived from the same source
causes problems, primarily that they can't contain hyperlinks
cross-referencing each other.

This crates dedicated private modules in `std` to document the
primitive types, then for all primitives that have a corresponding
module, puts hyperlinks in moth the primitive docs and the module docs
cross-linking each other.

This should help clear up confusion when readers find themselves on
the wrong page.

This also removes all the duplicate `#[doc(primitive)]` tags in various places (especially core), so the core docs will no longer attempt to document the primitives for now. Seems like an acceptable tradeoff to get some cleanup for std.

### CODE REVIEW

File: src/libstd/lib.rs [hunk 1]
```diff
@@ -415,12 +415,10 @@ pub mod __rand {
     pub use rand::{thread_rng, ThreadRng, Rng};
 }
 
-// Modules that exist purely to document + host impl docs for primitive types
-
-mod array;
-mod bool;
-mod unit;
-mod tuple;
+// Include a private number of modules that exist soley to provide the
```
Comment (apasel422) [inline src/libstd/lib.rs]: s/soley/solely/

File: src/libstd/num/f32.rs [hunk 2]
```diff
@@ -8,11 +8,13 @@
 // option. This file may not be copied, modified, or distributed
 // except according to those terms.
 
-//! Operations and constants for 32-bits floats (`f32` type)
+//! The 32-bit floating point type
```
Comment (steveklabnik) [inline src/libstd/num/f32.rs]: end with a .

File: src/libstd/num/f64.rs [hunk 3]
```diff
@@ -8,11 +8,12 @@
 // option. This file may not be copied, modified, or distributed
 // except according to those terms.
 
-//! Operations and constants for 64-bits floats (`f64` type)
+//! The 32-bit floating point type
```
Comment (steveklabnik) [inline src/libstd/num/f64.rs]: end with a .

File: src/libstd/num/i16.rs [hunk 4]
```diff
@@ -8,10 +8,11 @@
 // option. This file may not be copied, modified, or distributed
 // except according to those terms.
 
-//! Operations and constants for signed 16-bits integers (`i16` type)
+//! The 16-bit signed integer type
```
Comment (steveklabnik) [inline src/libstd/num/i16.rs]: i'll just leave it at that, all summaries should end with .

**Comment by steveklabnik** [comment]
A few nits but LGTM

File: src/libstd/num/f64.rs (hunk 3 above)
Comment (petrochenkov) [inline src/libstd/num/f64.rs]: Also
\>f64.rs
\>32-bit floating point type

**Comment by brson** [comment]
Feedback addressed. Thanks for review. r=steveklibnik

**Pull request merged**